geopandas/geoseries.py

```diff
@@ -189,14 +188,13 @@ def __init__(self, data=None, index=None, crs: Optional[Any] = None, **kwargs):
             # https://github.com/pandas-dev/pandas/issues/26469
             kwargs.pop("dtype", None)
             # Use Series constructor to handle input data
-            with compat.ignore_shapely2_warnings():
-                # suppress additional warning from pandas for empty data
-                # (will always give object dtype instead of float dtype in the future,
-                # making the `if s.empty: s = s.astype(object)` below unnecessary)
-                empty_msg = "The default dtype for empty Series"
-                warnings.filterwarnings("ignore", empty_msg, DeprecationWarning)
-                warnings.filterwarnings("ignore", empty_msg, FutureWarning)
-                s = pd.Series(data, index=index, name=name, **kwargs)
+            # suppress additional warning from pandas for empty data
+            # (will always give object dtype instead of float dtype in the future,
+            # making the `if s.empty: s = s.astype(object)` below unnecessary)
+            empty_msg = "The default dtype for empty Series"
+            warnings.filterwarnings("ignore", empty_msg, DeprecationWarning)
+            warnings.filterwarnings("ignore", empty_msg, FutureWarning)
+            s = pd.Series(data, index=index, name=name, **kwargs)
```

Comment:
Here I think you need to keep the `with` context, but using the plain `with warnings.catch_warnings():` (which is now used under the hood by `ignore_shapely2_warnings`), otherwise the filterwarnins are not applied only in this context.